server/src/main/java/org/elasticsearch/cluster/routing/allocation/allocator/DesiredBalanceReconciler.java

```diff
@@ -504,10 +524,27 @@ private void balance() {
                 }
             }
 
-            maybeLogUndesiredAllocationsWarning(allAllocations, undesiredAllocations, routingNodes.size());
+            DesiredBalanceReconciler.this.unassignedShards.set(unassignedShards);
+            DesiredBalanceReconciler.this.undesiredAllocations.set(undesiredAllocations);
+            DesiredBalanceReconciler.this.totalAllocations.set(totalAllocations);
+
+            logReconciliationMetrics.maybeExecute(
+                () -> logger.debug(
```

Comment:
You mentioned that the periodic logging has the purpose of creating a dashboard. Is the debug level good then? Meaning, we'd need to enable it for specific clusters then.

Reply:
Correct, we need to explicitly enable it. It is going to be disabled by default for users (to avoid adding more noise to the logs).